### PR TITLE
Implement Low Power mode support in CMIS devices.

### PR DESCRIPTION
**Pull request by aravindmani-1**
#### Description
Add LP mode support for CMIS devices.

#### Motivation and Context
Low Power mode can be set in CMIS devices by reading/writing in EEPROM.

#### How Has This Been Tested?
Tested in DellEMC Z9332f platform.
[CMIS lpmode.txt](https://github.com/Azure/sonic-platform-common/files/7496110/CMIS.lpmode.txt)

### CODE REVIEW

File: sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ def __init__(self, codes):
             )
         )
 
+        self.MODULE_STATE = NumberRegField(consts.MODULE_STATE_FIELD, self.getaddr(0x0, 3), size=1)
+        self.SET_LP_MODE = NumberRegField(consts.SET_LP_MODE_FIELD, self.getaddr(0x0, 26), size=1, ro=False)
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py:31]: can you reuse TRANS_CONFIG field?

File: sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py [hunk 2]
```diff
@@ -27,6 +27,9 @@ def __init__(self, codes):
             )
         )
 
+        self.MODULE_STATE = NumberRegField(consts.MODULE_STATE_FIELD, self.getaddr(0x0, 3), size=1)
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py:30]: can you reuse TRANS_MODULE_STATUS field?

**Review by prgeor — CHANGES_REQUESTED** [review]

File: sonic_platform_base/sonic_xcvr/api/public/cmis.py [hunk 3]
```diff
@@ -588,6 +588,29 @@ def get_lpmode_support(self):
...
+        if self.is_flat_memory() or not self.get_lpmode_support():
+            return False
+
+        lpmode = self.xcvr_eeprom.read(consts.MODULE_STATE_FIELD)
+        if lpmode is not None:
+            if lpmode >> 1 == 1:
+                return True
+        return False
+
+    def set_lpmode(self, lpmode):
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/api/public/cmis.py:601]: can you remove set_low_power()

File: sonic_platform_base/sonic_xcvr/api/public/cmis.py [hunk 4]
```diff
@@ -589,27 +589,31 @@ def get_lpmode_support(self):
...
 
-        lpmode_val = self.xcvr_eeprom.read(consts.MODULE_STATE_FIELD)
-        if lpmode_val is not None:
-            if lpmode is True:
-                lpmode_val = lpmode_val | (1 << 4)
-            else:
-                lpmode_val = lpmode_val & ~(1 << 4)
-            return self.xcvr_eeprom.write(consts.SET_LP_MODE_FIELD, lpmode_val)
-        return False
+        lpmode_val = lpmode << 4
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/api/public/cmis.py:615]: This is affecting other bits. This should affect only bit 4 LowPwrRequestSW

File: sonic_platform_base/sonic_xcvr/fields/consts.py [hunk 5]
```diff
@@ -100,8 +100,6 @@
 POWER_CTRL_FIELD = "Power Control"
 POWER_OVERRIDE_FIELD = "Power Override"
 POWER_SET_FIELD = "Power Set"
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/fields/consts.py:102]: can you remove changes in this file?

File: sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py [hunk 6]
```diff
@@ -27,9 +27,6 @@ def __init__(self, codes):
             )
         )
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py]: can you remove changes in this file

**Review by prgeor — CHANGES_REQUESTED** [review]

**Comment by aravindmani-1** [comment]
Will open a new PR to fix this issue.

**Comment by aravindmani-1** [comment]
https://github.com/Azure/sonic-platform-common/pull/237 will fix this issue.

**Pull request closed without merging**